Submission.java

```diff
@@ -210,6 +235,195 @@ private int getCountForStage(String item)
     }
 
     /**
+     * Count block categories for stage.
+     */
+    public void countBlockCategoriesForStage()
+    {
+        controlBlocksForStage = 0;
+        dataBlocksForStage = 0;
+        eventsBlocksForStage = 0;
+        looksBlocksForStage = 0;
+        moreBlocksBlocksForStage = 0;
+        motionBlocksForStage = 0;
+        operatorsBlocksForStage = 0;
+        penBlocksForStage = 0;
+        sensingBlocksForStage = 0;
+        soundBlocksForStage = 0;
+        JSONArray scripts = FileUtils.getJSONArrayAttribute(jsonObj, "scripts");
+        processScripts(scripts);
+    }
+
+    /**
+     * Process scripts to count blocks by category.
+     *
+     * @param array 
+     */
+    private void processScripts(JSONArray array)
+    {
+        if (array == null || array.size() == 0)
+        {
+            return;
+        }
+        
+        // If first element is a String, it is the block name.
+        // Get and count its category.
+        if (array.get(0) instanceof String)
+        {
+            String category = getCategory((String) array.get(0));
+            if (category != null)
```

Comment:
just have a question.. why do you have to typecast array.get(0) to a string if the condition array.get(0) instanceof String evaluates to true?